components/user-store/org.wso2.carbon.identity.user.store.configuration/src/main/java/org/wso2/carbon/identity/user/store/configuration/utils/SecondaryUserStoreConfigurationUtil.java

```diff
@@ -778,6 +779,77 @@ public static void triggerListenersOnUserStorePreAdd(String domainName) throws U
         }
     }
 
+    /**
+     * Trigger the listeners after a userstore is retrieved.
+     *
+     * @param userStoreDTO Retrieved userstore configuration.
+     * @throws UserStoreException throws when an error occurred when triggering listeners.
+     */
+    public static void triggerListenersOnUserStorePostGet(UserStoreDTO userStoreDTO) throws UserStoreException {
+
+        int tenantId = CarbonContext.getThreadLocalCarbonContext().getTenantId();
+        List<UserStoreConfigListener> userStoreConfigListeners = UserStoreConfigListenersHolder.getInstance()
+                .getUserStoreConfigListeners();
+
+        for (UserStoreConfigListener userStoreConfigListener : userStoreConfigListeners) {
```

Comment:
isn't it worth to have a debug log denoting userstore listeners that were executed ?

Reply:
added